### PR TITLE
feat: add eslint, change  grid and tabs to be generic

### PR DESCRIPTION
**Pull request by NiklasLimberg**
This PR adds tons of ESLint rules to give the frontend code a uniform look and feel.

It also makes the tabs and the grid generic, to provide autocomplete for slots and slot data

### CODE REVIEW

File: frontend/src/components/layout/Tabs.vue [hunk 1]
```diff
@@ -1,46 +1,78 @@
...
+                        <component
+                            :is="label.icon"
+                            v-if="label.icon"
+                            class="mr-2"
+                        />
                         {{ label.title }}
-                        <span v-if="label.count !== undefined"
-                            class="ml-2 bg-gray-300 rounded-full px-2.5 py-0.5 text-xs font-medium text-gray-900">
+                        <span
+                            v-if="label.count !== undefined"
```
Comment (pumpi) [inline frontend/src/components/layout/Tabs.vue:32]: Label counts are gone
Bildschirmfoto 2024-03-09 um 18 48 58
Bildschirmfoto 2024-03-09 um 18 49 12

File: frontend/src/views/account/ListExtensions.vue [hunk 2]
```diff
@@ -1,6 +1,175 @@
...
+                        class="-ml-1 mr-2 h-4 w-4 opacity-25 group-hover:opacity-50"
+                        aria-hidden="true"
+                    />
+                    Add Shop
+                </router-link>
+            </div>
+        </div>
+
+        <template v-else>
+            <input
```
Comment (pumpi) [inline frontend/src/views/account/ListExtensions.vue:45]: Search no longer works
Bildschirmfoto 2024-03-09 um 18 39 24
Bildschirmfoto 2024-03-09 um 18 39 55

File: frontend/src/views/account/ListExtensions.vue [hunk 3]
```diff
@@ -1,6 +1,175 @@
...
+                    :columns="[
+                        { key: 'label', name: 'Name', sortable: true },
+                        { key: 'shops', name: 'Shop' },
+                        { key: 'version', name: 'Version' },
+                        { key: 'latestVersion', name: 'Latest' },
+                        { key: 'ratingAverage', name: 'Rating', sortable: true },
+                        { key: 'installedAt', name: 'Installed At', sortable: true },
+                    ]"
+                    :data="extensionStore.extensions"
+                    :default-sorting="{by: 'label'}"
```
Comment (pumpi) [inline frontend/src/views/account/ListExtensions.vue:65]: Default sorting no longer works

File: frontend/src/views/shop/DetailShop.vue [hunk 4]
```diff
@@ -0,0 +1,903 @@
...
+                            target="_blank"
+                        >
+                            {{ row.message }}
+                            <icon-fa6-solid:up-right-from-square class="text-xs" />
+                        </a>
+                        <template v-else>
+                            {{ row.message }}
+                        </template>
+                    </template>
+                    <template #cell-actions="{ row }">
```
Comment (pumpi) [inline frontend/src/views/shop/DetailShop.vue:227]: Action label is gone
Bildschirmfoto 2024-03-09 um 18 50 14
Bildschirmfoto 2024-03-09 um 18 50 26

Bildschirmfoto 2024-03-09 um 18 55 11
Bildschirmfoto 2024-03-09 um 18 55 22

File: frontend/src/views/shop/DetailShop.vue [hunk 5]
```diff
@@ -0,0 +1,903 @@
...
+                <data-table
+                    :columns="[
+                        { key: 'name', name: 'Name', sortable: true },
+                        { key: 'size', name: 'Size', sortable: true },
+                    ]"
+                    :data="shopStore.shop.queueInfo || []"
+                />
+            </template>
+
+            <template #panel-pagespeed>
```
Comment (pumpi) [inline frontend/src/views/shop/DetailShop.vue:424]: Change to tab page speed get an JS error
Bildschirmfoto 2024-03-09 um 19 03 16
Bildschirmfoto 2024-03-09 um 19 03 37

File: frontend/src/components/layout/NavBar.vue
Comment (pumpi) [inline frontend/src/components/layout/NavBar.vue:1]: Please rebase from my last Changes from Nav.vue where I change the icon layout of nav and add Github icon

File: frontend/src/components/layout/Tabs.vue (hunk 1 above)
Comment (NiklasLimberg) [reply frontend/src/components/layout/Tabs.vue:32]: Just forgot this in the column definition

File: frontend/src/views/account/ListExtensions.vue (hunk 2 above)
Comment (NiklasLimberg) [reply frontend/src/views/account/ListExtensions.vue:45]: Fixed that by marking name as searchable

File: frontend/src/views/shop/DetailShop.vue (hunk 5 above)
Comment (NiklasLimberg) [reply frontend/src/views/shop/DetailShop.vue:454]: fixed this too

File: frontend/src/components/layout/NavBar.vue
Comment (NiklasLimberg) [reply frontend/src/components/layout/NavBar.vue:1]: I copied over the HTML from origin/main

File: frontend/src/views/account/ListExtensions.vue (hunk 3 above)
Comment (NiklasLimberg) [reply frontend/src/views/account/ListExtensions.vue]: Readded this even tho this technically bad practice

File: frontend/src/views/shop/DetailShop.vue (hunk 4 above)
Comment (NiklasLimberg) [reply frontend/src/views/shop/DetailShop.vue:254]: Added a slot for the action label and filled it in these cases

File: frontend/src/views/shop/ListShops.vue [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+            </div>
+        </div>
+
+        <div
+            v-else
+            class="shadow rounded-md overflow-y-scroll md:overflow-y-hidden bg-white dark:bg-neutral-800 dark:shadow-none"
+        >
+            <data-table
+                :columns="[
+                    { key: 'favicon', name: 'Logo' },
```
Comment (pumpi) [inline frontend/src/views/shop/ListShops.vue:61]: The favicon col has no title and has other styles before
`:labels="{favicon: {name: '', classOverride: true, class: 'w-11 min-w-[44px] py-3.5 px-3'}`
image
image

**Review by pumpi — COMMENTED** [review]
One last thing I found then we can merge I think ;)

File: frontend/src/views/shop/ListShops.vue (hunk 6 above)
Comment (NiklasLimberg) [reply frontend/src/views/shop/ListShops.vue]: fixed it with https://github.com/FriendsOfShopware/shopmon/compare/432eb466bf4543ca446cd2346cc30eb2b08145ac..5e53a5554f16e332639b661c1d125be2c9f14dfb#diff-8275cc51c7438defe929163bde969ba57fc5bc1836a132a7f579b28bdc6efca2R61

**Pull request merged**